pkg/errinsrc/internal/golocation.go

```diff
@@ -57,3 +63,40 @@ func FromGoTokenPositions(start token.Position, end token.Position) *SrcLocation
 		Type:  LocError,
 	}
 }
+
+// convertSingleGoPositionToRange attempts to convert a single Go token position to a range with a start and end
+// position.
+//
+// This is done by attempting to parse the file, and then if we are able to parse it successfully, we look for an AST
+// node which starts at the exact line and column of the position. If we find one, we use the end position of that
+// node as the end position of the range.
+//
+// We use the first found node at that position, as we assume the largest node at that position is the most relevant
+func convertSingleGoPositionToRange(filename string, fileBody []byte, start token.Position) (end token.Position) {
+	fs := token.NewFileSet()
+	file, err := parser.ParseFile(fs, filename, fileBody, parser.ParseComments)
+
+	if err == nil && file != nil {
+		var match ast.Node
+
+		ast.Inspect(file, func(n ast.Node) bool {
```

Comment:
I think this would return the smallest matching node; where as I'm returning the largest matching node right now - which I think is the behaviour we want.

i.e.

```
x := foo.Bar("hello")
     ^   ^
     |   ` we want the *ast.Ident
     ` we want the *ast.CallExpr - not the *ast.Sel ("foo.Bar") and not the *ast.Ident ("foo")
```

Reply:
`PathEnclosingInterval` returns all of them, in order (from smallest to largest), I believe.

Reply:
I read it as;
> and all its ancestors up to the AST root.

Which isn't the same as smallest match to largest.

For now I know what we've got works so don't want to swap it out and have to retest a bunch of senarios